File: backend/app/schemas/classification.py
from pydantic import BaseModel
from uuid import UUID
from datetime import datetime

class SpeciesResultOut(BaseModel):
  species_id: UUID
  scientific_name: str
  score: float


class ClassificationOut(BaseModel):
  classification_id: UUID
  classification_date: datetime
  original_image_url: str
  location: str | None
  predictions: list[SpeciesResultOut]

  class Config:
    orm_mode = True


class PredictionResult(BaseModel):
  class_id: int
  label: str
  confidence: float


class ClassificationResultOut(BaseModel):
  classification_id: UUID
  top_k: int
  predictions: list[PredictionResult]


class RecentClassificationOut(BaseModel):
  classification_id: UUID
  classification_date: datetime
  original_image_url: str
  location: str | None
  top_prediction: SpeciesResultOut

File: frontend/types/classification.ts
export type Prediction = {
  class_id: number
  label: string
  confidence: number
}

export type ClassificationResultResponse = {
  classification_id: string
  top_k: number
  predictions: Prediction[]
}

export type SpeciesResult = {
  species_id: string
  scientific_name: string
  score: number
}

export type ClassificationResponse = {
  classification_id: string
  classification_date: string
  original_image_url: string
  location: string | null
  predictions: SpeciesResult[]
}
